e2e/specs/drug-orders.spec.ts

```diff
@@ -0,0 +1,185 @@
+import { expect } from '@playwright/test';
+import { type Visit } from '@openmrs/esm-framework';
+import { generateRandomPatient, deletePatient, type Patient, startVisit, endVisit } from '../commands';
+import { test } from '../core';
+import { MedicationsPage } from '../pages';
+
+let patient: Patient;
+let visit: Visit;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+  visit = await startVisit(api, patient.uuid);
+});
+
+test('Record, edit and discontinue a drug order', async ({ page }) => {
+  const medicationsPage = new MedicationsPage(page);
+
+  await test.step('When I visit the medications page', async () => {
+    await medicationsPage.goTo(patient.uuid);
+  });
+
+  await test.step('And I click on the `Record active medications` link', async () => {
+    await page.getByText(/record active medications/i).click();
+  });
+
+  await test.step('And I search for the drug in the search bar', async () => {
+    await page.getByRole('searchbox', { name: /search for a drug or orderset/i }).fill('aspirin');
+  });
+
+  await test.step('And I click on the `Add to basket` button', async () => {
+    await page
+      .getByRole('listitem')
+      .filter({ hasText: /aspirin 81mg — 81mg — tablet/i })
+      .getByRole('button', { name: /add to basket/i })
+      .click();
+  });
+
+  await test.step('And I click on the drug order form link', async () => {
+    await page
+      .getByRole('listitem')
+      .filter({ hasText: /incomplete/i })
+      .click();
+  });
+
+  await test.step('Then I should be redirected to the drug order form', async () => {
+    await expect(page.getByText(/order form/i)).toBeVisible();
+  });
+
+  await test.step('When I set the does to `1` tablet', async () => {
+    await medicationsPage.page.getByLabel(/^dose$/i).clear();
```

Comment:
```suggestion
    await page.getByLabel(/^dose$/i).clear();
```

Both page and medicationPage.page share the same instance. Let's use page for clarity's sake.